ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala

```diff
@@ -74,51 +74,55 @@ class IndexDBMetrics(override val prefix: MetricName, override val registry: Met
     "lookup_contract_by_key"
   )
 
-  // TODO etq: Tune metric names
-  // TODO etq: Document new metrics and changes to the existing metrics
-  val flatTxIdsCreate: DatabaseMetrics = createDbMetrics(
-    "get_create_event_ids_for_flat_transactions"
-  )
-  val flatTxIdsConsuming: DatabaseMetrics = createDbMetrics(
-    "get_consuming_event_ids_for_flat_transactions"
-  )
-  val flatTxPayloadCreate: DatabaseMetrics = createDbMetrics(
-    "get_create_event_payloads_for_flat_transactions"
-  )
-  val flatTxPayloadConsuming: DatabaseMetrics = createDbMetrics(
-    "get_consuming_event_payloads_for_flat_transactions"
-  )
-  val treeTxIdsCreateStakeholder: DatabaseMetrics = createDbMetrics(
-    "get_create_event_ids_for_stakeholder_for_tree_transactions"
-  )
-  val treeTxIdsCreateNonStakeholderInformee: DatabaseMetrics = createDbMetrics(
-    "get_create_event_ids_for_nonstakeholderinformee_for_tree_transactions"
-  )
-  val treeTxIdsConsumingStakeholder: DatabaseMetrics = createDbMetrics(
-    "get_consuming_event_ids_for_stakeholder_for_tree_transactions"
-  )
-  val treeTxIdsConsumingNonStakeholderInformee: DatabaseMetrics = createDbMetrics(
-    "get_consuming_event_ids_for_nonstakeholderinformee_for_tree_transactions"
-  )
-  val treeTxIdsNonConsumingInformee: DatabaseMetrics = createDbMetrics(
-    "get_nonconsuming_event_ids_for_informee_for_tree_transactions"
-  )
-  val treeTxPayloadCreate: DatabaseMetrics = createDbMetrics(
-    "get_create_event_payloads_for_tree_transactions"
-  )
-  val treeTxPayloadConsuming: DatabaseMetrics = createDbMetrics(
-    "get_consuming_event_payloads_for_tree_transactions"
-  )
-  val treeTxPayloadNonConsuming: DatabaseMetrics = createDbMetrics(
-    "get_nonconsuming_event_payloads_for_tree_transactions"
-  )
-  // TODO etq: Document new metrics and changes to the existing metrics
-  val getFlatTransactions: DatabaseMetrics = createDbMetrics("get_flat_transactions")
+  object flatTxStream {
+    private val prefix = "flat_transactions_stream"
+    val fetchEventCreateIdsStakeholder: DatabaseMetrics = createDbMetrics(
+      prefix + "_fetch_event_create_ids_stakeholder"
+    )
+    val fetchEventConsumingIdsStakeholder: DatabaseMetrics = createDbMetrics(
+      prefix + "_fetch_event_consuming_ids_stakeholder"
+    )
+    val fetchEventCreatePayloads: DatabaseMetrics = createDbMetrics(
+      prefix + "_fetch_event_create_payloads"
+    )
+    val fetchEventConsumingPayloads: DatabaseMetrics = createDbMetrics(
+      prefix + "_fetch_event_consuming_payloads"
+    )
+    val translationTimer: Timer = createDbMetrics(prefix).translationTimer
+  }
+
+  object treeTxStream {
+    private val prefix = "tree_transactions_stream"
+    val fetchEventCreateIdsStakeholder: DatabaseMetrics = createDbMetrics(
+      prefix + "_fetch_event_create_ids_stakeholder"
```

Comment:
object treeTxStream extends FactoryWithDBMetrics {
    override val prefix: MetricName = self.prefix :+ "tree_transactions_stream"
    override val registry: MetricRegistry = self.registry
    val fetchEventCreateIdsStakeholder: DatabaseMetrics = createDbMetrics(
      "fetch_event_create_ids_stakeholder"
```
I would do it like this so that there is a segment delimiter between `tree_transactions_stream` and individual metrics. Such delimiters get different translations in different back-ends

Reply:
Done.
A side-effect of they way I applied your suggestion is that we'll have a `daml_index_db_flat_transactions_stream_translation_translation` metric